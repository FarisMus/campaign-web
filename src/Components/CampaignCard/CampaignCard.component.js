import React from 'react';

import styles from './CampaignCard.component.styles';

const _renderImage = (src) => {
	return (
		<div style={styles.image}>
			<img src={src} alt={'poster'} width={'100%'}/>
		</div>
	);
};

const _renderTitle = (title) => {
	return <div style={styles.title}>{title}</div>;
};

const _renderIncompleteProgressBar = (percentage) => {
	const progress = percentage * 100;
	const rest = 100 - progress;

	return (
		<div style={styles.progressBarContainer}>
			<div style={{
				border: '#636363 8px solid',
				borderRadius: '8px 0 0 8px',
				width: `${progress}%`
			}}/>
			<div style={{
				border: '#dbdbdb 8px solid',
				borderRadius: '0 8px 8px 0',
				width: `${rest}%`
			}}
			/>
		</div>
	);
};

const _renderProgressBar = (percentage) => {
	if(percentage < 1.0) {
		return _renderIncompleteProgressBar(percentage);
	}

	return (
		<div style={styles.progressBarContainer}>
			<div style={{
				border: '#ffa7f3 8px solid',
				borderRadius: '8px 8px 8px 8px',
				width: `${percentage * 100}%`
			}}/>
		</div>
	)
};

const _renderInfo = (donationReceived, daysRemaining) => {
	const currencyFormatter = new Intl.NumberFormat('id-ID', {
		style: 'currency',
		currency: 'IDR'
	});
	const formattedAmount = currencyFormatter.format(donationReceived);

	return (
		<div>
			<div style={styles.infoRow}>
				<div>
					Terkumpul
				</div>
				<div>
					Sisa hari
				</div>
			</div>
			<div style={styles.infoRow}>
				<div>
					{formattedAmount}
				</div>
				<div>
					{daysRemaining}
				</div>
			</div>
		</div>
	);
};

const CampaignCard = (props) => {
	return (
		<div style={styles.card}>
			{_renderImage(props.image)}
			{_renderTitle(props.title)}
			{_renderProgressBar(props.donation_percentage)}
			{_renderInfo(props.donation_target, props.days_remaining)}
		</div>
	);
};

export default CampaignCard;
