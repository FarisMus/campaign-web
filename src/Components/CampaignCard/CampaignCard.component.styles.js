export default {
	image: {
		width: '100%',
		height: '18em'
	},
	title: {
		fontWeight: 'bold',
		fontSize: 20,
		margin: '0.5em 0 0.5em 0',
		whiteSpace: 'nowrap',
		overflow: 'hidden',
		textOverflow: 'ellipsis',
		padding: '0 0.5em 0 0.5em'
	},
	progressBarContainer: {
		display: 'flex',
		flexDirection: 'row',
		padding: '0 0.5em 0 0.5em'
	},
	infoRow: {
		display: 'flex',
		flexDirection: 'row',
		justifyContent: 'space-between',
		padding: '0.5em 0.5em 0 0.5em'
	},
	card: {
		width: '33em',
		paddingBottom: '2em',
		border: 'solid #d4d4d4 1px',
		boxShadow: '4px 4px 2px #d4d4d4',
		borderRadius: '4px 4px 4px 4px',
		margin: '0.5em',
	}
};
