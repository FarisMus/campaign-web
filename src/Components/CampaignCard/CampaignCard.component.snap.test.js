import React from 'react';
import renderer from 'react-test-renderer';
import CampaignCard from './CampaignCard.component';

describe('Render Campaign Card Component', () => {
	const defaultProps = {
		image: 'image.png',
		title: 'Campaign Title',
		donation_percentage: 0.02,
		donation_target: 1000000,
		days_remaining: 32
	};

	test('Render CampaignCard with unreached target', () => {
		const component = renderer.create(
			<CampaignCard {...defaultProps} />,
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Render CampaignCard with reached target', () => {
		const _props = {
			...defaultProps,
			donation_percentage: 1.0
		};
		const component = renderer.create(
			<CampaignCard {..._props} />,
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
