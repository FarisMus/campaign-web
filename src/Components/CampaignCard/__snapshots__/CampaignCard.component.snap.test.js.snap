// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Campaign Card Component Render CampaignCard with reached target 1`] = `
<div
  style={
    Object {
      "border": "solid #d4d4d4 1px",
      "borderRadius": "4px 4px 4px 4px",
      "boxShadow": "4px 4px 2px #d4d4d4",
      "margin": "0.5em",
      "paddingBottom": "2em",
      "width": "33em",
    }
  }
>
  <div
    style={
      Object {
        "height": "18em",
        "width": "100%",
      }
    }
  >
    <img
      alt="poster"
      src="image.png"
      width="100%"
    />
  </div>
  <div
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "margin": "0.5em 0 0.5em 0",
        "overflow": "hidden",
        "padding": "0 0.5em 0 0.5em",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    Campaign Title
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "padding": "0 0.5em 0 0.5em",
      }
    }
  >
    <div
      style={
        Object {
          "border": "#ffa7f3 8px solid",
          "borderRadius": "8px 8px 8px 8px",
          "width": "100%",
        }
      }
    />
  </div>
  <div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "0.5em 0.5em 0 0.5em",
        }
      }
    >
      <div>
        Terkumpul
      </div>
      <div>
        Sisa hari
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "0.5em 0.5em 0 0.5em",
        }
      }
    >
      <div>
        IDR 1,000,000.00
      </div>
      <div>
        32
      </div>
    </div>
  </div>
</div>
`;

exports[`Render Campaign Card Component Render CampaignCard with unreached target 1`] = `
<div
  style={
    Object {
      "border": "solid #d4d4d4 1px",
      "borderRadius": "4px 4px 4px 4px",
      "boxShadow": "4px 4px 2px #d4d4d4",
      "margin": "0.5em",
      "paddingBottom": "2em",
      "width": "33em",
    }
  }
>
  <div
    style={
      Object {
        "height": "18em",
        "width": "100%",
      }
    }
  >
    <img
      alt="poster"
      src="image.png"
      width="100%"
    />
  </div>
  <div
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "margin": "0.5em 0 0.5em 0",
        "overflow": "hidden",
        "padding": "0 0.5em 0 0.5em",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    Campaign Title
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "padding": "0 0.5em 0 0.5em",
      }
    }
  >
    <div
      style={
        Object {
          "border": "#636363 8px solid",
          "borderRadius": "8px 0 0 8px",
          "width": "2%",
        }
      }
    />
    <div
      style={
        Object {
          "border": "#dbdbdb 8px solid",
          "borderRadius": "0 8px 8px 0",
          "width": "98%",
        }
      }
    />
  </div>
  <div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "0.5em 0.5em 0 0.5em",
        }
      }
    >
      <div>
        Terkumpul
      </div>
      <div>
        Sisa hari
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "0.5em 0.5em 0 0.5em",
        }
      }
    >
      <div>
        IDR 1,000,000.00
      </div>
      <div>
        32
      </div>
    </div>
  </div>
</div>
`;
